Gems/Atom/Feature/Common/Code/Source/FrameCaptureSystemComponent.cpp

```diff
@@ -389,94 +389,80 @@ namespace AZ
+                capture->m_outputFilePath = ResolvePath(imagePath);
             }
-            capture->m_readback->SetUserIdentifier(captureHandle.GetCaptureStateIndex());
-            capture->m_readback->SetCallback(AZStd::bind(&FrameCaptureSystemComponent::CaptureAttachmentCallback, this, AZStd::placeholders::_1));
 
-            capture->m_outputFilePath = ResolvePath(filePath);
-            pass->ReadbackSwapChain(capture->m_readback);
-            m_inProgressCaptures.push_back(captureHandle);
+            return captureHandle;
+        }
 
-            return captureHandle.GetCaptureStateIndex();
+        FrameCaptureId FrameCaptureSystemComponent::CaptureScreenshotForWindow(const AZStd::string& filePath, AzFramework::NativeWindowHandle windowHandle)
+        {
+            return InternalCaptureScreenshot(filePath, windowHandle);
         }
 
         FrameCaptureId FrameCaptureSystemComponent::CaptureScreenshot(const AZStd::string& filePath)
         {
             AzFramework::NativeWindowHandle windowHandle = AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContext()->GetWindowHandle();
             if (windowHandle)
             {
-                return CaptureScreenshotForWindow(filePath, windowHandle);
+                return InternalCaptureScreenshot(filePath, windowHandle);
             }
 
             return InvalidFrameCaptureId;
         }
 
         FrameCaptureId FrameCaptureSystemComponent::CaptureScreenshotWithPreview(const AZStd::string& outputFilePath)
         {
-            if (!CanCapture())
-            {
-                return InvalidFrameCaptureId;
-            }
-
-            CaptureHandle captureHandle = InitCapture();
+            CaptureHandle captureHandle = ScreenshotPreparation(outputFilePath, nullptr);
             if (captureHandle.IsNull())
             {
-                AZ_Assert(false, "Failed to allocate a capture");
                 return InvalidFrameCaptureId;
             }
             AZStd::scoped_lock<CaptureHandle> scope_lock(captureHandle);
-            CaptureState* capture = captureHandle.GetCaptureState();
-            if (!capture) // failed to get the capture state ptr, abort
-            {
-                m_idleCaptures.push_back(captureHandle);
-                return InvalidFrameCaptureId;
-            }
-
-            if (!capture->m_readback->IsReady())
-            {
-                AZ_Assert(false, "Failed to capture attachment since the readback is not ready");
-                m_idleCaptures.push_back(captureHandle);
-                return InvalidFrameCaptureId;
-            }
-            capture->m_readback->SetUserIdentifier(captureHandle.GetCaptureStateIndex());
-            capture->m_readback->SetCallback(AZStd::bind(&FrameCaptureSystemComponent::CaptureAttachmentCallback, this, AZStd::placeholders::_1));
-
-            capture->m_outputFilePath = ResolvePath(outputFilePath);
+            CaptureState* captureState = captureHandle.GetCaptureState();
```

Comment:
I think these can be moved later in the function so it's not holding the lock as long.